src/common/include/sirf/common/GeometricalInfo.h

```diff
@@ -0,0 +1,173 @@
+/*
+  CCP PETMR Synergistic Image Reconstruction Framework (SIRF)
+  Copyright 2018 Commonwealth Scientific and Industrial Research Organisation's
+  Australian eHealth Research Organisation
+
+  This is software developed for the Collaborative Computational
+  Project in Positron Emission Tomography and Magnetic Resonance imaging
+  (http://www.ccppetmr.ac.uk/).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#ifndef SIRF_GEOMETRICAL_INFO_TYPE
+#define SIRF_GEOMETRICAL_INFO_TYPE
+
+namespace sirf {
+
+template <int num_dimensions, typename T>
+class tVector {
```

Comment:
not sure about the name here...

I guess we do need a class for fixed-length arrays. Any reason we're not using `std::array`?